### DISCUSSION TITLE
simbot启动问题

### DISCUSSION THREAD

**Discussion by star4277** [simple-robot/Discussions#148] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

我现在已经将插件弄得差不多了，我现在想问一下如何在使用@EnableSimbot注解的条件下启动springboot项目，停止初始化simbot的application所有初始化，因为主程序和插件同时启动同一个机器人，会导致互相挤掉线，是只能自己写starter了吗

**Comment by ForteScarlet** [comment]
`Application` 的注册是从 [SimbotSpringBootApplicationConfiguration.kt#L76-L107](https://github.com/simple-robot/simpler-robot/blob/7940e7032b15dd39cc9a693d07371b7f150acdec/simboot-core-spring-boot-starter/src/main/kotlin/love/forte/simboot/spring/autoconfigure/SimbotSpringBootApplicationConfiguration.kt#L76-L107) 进行的。可以选择自行注册一个 `Application` 来替代它（因为有 `@ConditionalOnMissingBean`）或者想办法排除掉它

而且既然是插件化，插件应当只是对处理逻辑的管理（简单来说就是对监听函数的管理），而不应是会启动bot的，这样也可以避免bot的频繁重启而导致一些问题，比如被风控。
